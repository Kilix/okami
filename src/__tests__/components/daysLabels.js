import React from 'react'
import {mount} from 'enzyme'
import toJson from 'enzyme-to-json'
import frLocale from 'date-fns/locale/fr'
import mockdate from 'mockdate'

import DaysLabels from '../../components/daysLabels'

describe('DaysLabels', () => {
  beforeEach(() => {
    mockdate.set('3/3/2017', 0)
  })
  afterEach(() => {
    mockdate.reset()
  })

  const ctx = {
    locale: frLocale,
    dateFormat: 'DD',
    showWeekend: true,
    rowHeight: 30,
    startWeek: new Date(2017, 9, 10, 0, 0, 0, 0),
  }
  test('render', () => {
    const tree = mount(<DaysLabels />, {context: ctx})
    expect(tree.html()).toMatchSnapshot()
  })
  test('render Child', () => {
    const tree = mount(<DaysLabels renderChild={({idx, ...props}) => <span {...props} />} />, {
      context: ctx,
    })
    expect(tree.html()).toMatchSnapshot()
  })
  test('function as a Child', () => {
    const tree = mount(
      <DaysLabels>
        {({weeks}) => <div>{weeks.map((w, idx) => <span key={idx}>{w}</span>)}</div>}
      </DaysLabels>,
      {context: ctx}
    )
    expect(tree.html()).toMatchSnapshot()
  })
  test('render with no weekend', () => {
    const tree = mount(<DaysLabels />, {context: {...ctx, showWeekend: false}})
    expect(tree.html()).toMatchSnapshot()
  })
})
