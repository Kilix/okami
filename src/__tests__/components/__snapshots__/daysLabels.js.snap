// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaysLabels function as a Child 1`] = `"<div><span>09</span><span>10</span><span>11</span><span>12</span><span>13</span><span>14</span><span>15</span></div>"`;

exports[`DaysLabels render 1`] = `"<div><div style=\\"height: 30px;\\">09</div><div style=\\"height: 30px;\\">10</div><div style=\\"height: 30px;\\">11</div><div style=\\"height: 30px;\\">12</div><div style=\\"height: 30px;\\">13</div><div style=\\"height: 30px;\\">14</div><div style=\\"height: 30px;\\">15</div></div>"`;

exports[`DaysLabels render Child 1`] = `"<div><span>09</span><span>10</span><span>11</span><span>12</span><span>13</span><span>14</span><span>15</span></div>"`;

exports[`DaysLabels render with no weekend 1`] = `"<div><div style=\\"height: 30px;\\">09</div><div style=\\"height: 30px;\\">10</div><div style=\\"height: 30px;\\">11</div><div style=\\"height: 30px;\\">12</div><div style=\\"height: 30px;\\">13</div></div>"`;
